### ISSUE TITLE
helm lint steps

### ISSUE THREAD

**Issue by VihanAgarwal97** [helm/helm#7930]
Is there a way to lint the file in steps? I'm using a for loop in the chart and so I'm not able to determine the exact line where the lint is failing.

Output of `helm version`:
```
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

**Comment by bacongobbler** [comment]
this is not currently possible, as far as I understand. There is a feature request in the queu asking for pluggable rulesets for `helm lint`. I'll see if I can find that. Would that help in this regard?

If you could elaborate more on your use case, that would be much appreciated.

**Comment by bacongobbler** [comment]
https://github.com/helm/helm/issues/2694

**Comment by VihanAgarwal97** [comment]
Thanks for the quick response. I'm using this with prometheus and have the following `configmap.yaml` -

```
receivers:
{{- range .Values.receivers }}
  - name: {{ .name }}
    pagerduty_configs:
      - routing_key: {{ .routing_key }}
        {{- if .send_resolved }}
        send_resolved: {{ .send_resolved }}
        {{- end }}
        {{- if .use_grouplabel_severity }}
        severity: {{ `{{ if .GroupLabels.severity }}{{ .GroupLabels.severity }}{{ else}}warning{{end}}` }}
        {{- else }}
        severity: {{ .default_severity }}
        {{- end }}
        description: {{` {{ template "slack.sezzle.title" . }} `}}
        links:
          - href: {{` {{ template "slack.actions.url" . }} `}}
            text: URL
{{- end }}
```

When I run this, I get an error with a line number - 
```
error converting YAML to JSON: yaml: line <x>: did not find expected key
```
But since this is a for loop, I'm finding it difficult to see which exact line this is.

**Comment by bacongobbler** [comment]
Ah, gotcha! Thank you for sharing your use case. I really appreciate it.

OK, so the line numbers are calculated from the fully-rendered YAML manifests *after* they've run through the template engine. In Helm 3.1, it is not possible to view the fully-rendered manifests if they're invalid due to #7088.

That being said, #7556 introduced the ability to dump the invalid YAML manifests with `helm template --debug`, which will be available in Helm 3.2.0. If you want to try [the canary release](https://helm.sh/docs/intro/install/#from-canary-builds), you can use `helm template --debug` to try and nail down which line is causing the issue.

Hope this helps!

**Comment by VihanAgarwal97** [comment]
This helps in my case. Not sure if the feature of stepping through linting might be in the future. But closing this issue irrespective. Thanks!!
